experimental/LieAlgebras/src/RootSystem.jl

```diff
@@ -1462,6 +1462,69 @@ function tensor_product_decomposition(
   )
 end
 
+###############################################################################
+#demazures charcter formula
+function _demazure_operator(r::RootSpaceElem, w::WeightLatticeElem)
+  fl, index_of_r = is_simple_root_with_index(r)
+  @req fl "not a simple root"
+  
+  d = 2*dot(w, r)//dot(r, r)
+  list_of_occuring_weights = WeightLatticeElem[]
+  
+  refl = reflect(w, index_of_r)
+
+  
+  if d > -1
+    while w != refl
+      push!(list_of_occuring_weights, w)
+      w -= WeightLatticeElem(r)
```

Comment:
you are computing `WeightLatticeElem(r)` over and over again. Could you change this function to instead compute it only once in the beginning, put it in a variable and then use that instead?